plugins/continuous-toolbox/src/ContinuousFlyout.ts

```diff
@@ -0,0 +1,292 @@
+      this.getParentToolbox().getCategoryByName(element.getButtonText())
+    );
+  }
+
+  /**
+   * Selects an item in the toolbox based on the scroll position of the flyout.
+   *
+   * @param position Current scroll position of the workspace.
+   */
+  private selectCategoryByScrollPosition(position: number) {
+    // If we are currently auto-scrolling, due to selecting a category by
+    // clicking on it, do not update the category selection.
+    if (this.scrollTarget) {
+      return;
+    }
+    const scaledPosition = Math.round(position / this.getWorkspace().scale);
+    // Traverse the array of scroll positions in reverse, so we can select the
+    // furthest category that the scroll position is beyond.
+    for (const [name, position] of [
+      ...this.scrollPositions.entries(),
+    ].reverse()) {
+      if (scaledPosition >= position) {
+        this.getParentToolbox().selectCategoryByName(name);
+        return;
+      }
+    }
+  }
+
+  /**
+   * Scrolls the flyout to given position.
+   *
+   * @param position The Y coordinate to scroll to.
+   */
+  scrollTo(position: number) {
+    // Set the scroll target to either the scaled position or the lowest
+    // possible scroll point, whichever is smaller.
+    const metrics = this.getWorkspace().getMetrics();
+    this.scrollTarget = Math.min(
+      position * this.getWorkspace().scale,
+      metrics.scrollHeight - metrics.viewHeight,
+    );
+
+    this.stepScrollAnimation();
+  }
+
+  /**
+   * Scrolls the flyout to display the given category at the top.
+   *
+   * @param category The toolbox category to scroll to in the flyout.
+   */
+  scrollToCategory(category: Blockly.ISelectableToolboxItem) {
+    const position = this.scrollPositions.get(category.getName());
+    if (!position) {
+      console.warn(`Scroll position not recorded for category ${name}`);
+      return;
+    }
+    this.scrollTo(position);
+  }
+
+  /**
+   * Step the scrolling animation by scrolling a fraction of the way to
+   * a scroll target, and request the next frame if necessary.
+   */
+  private stepScrollAnimation() {
+    if (!this.scrollTarget) return;
+
+    const currentScrollPos = -this.getWorkspace().scrollY;
+    const diff = this.scrollTarget - currentScrollPos;
+    if (Math.abs(diff) < 1) {
+      this.getWorkspace().scrollbar?.setY(this.scrollTarget);
+      this.scrollTarget = undefined;
+      return;
+    }
+    this.getWorkspace().scrollbar?.setY(
+      currentScrollPos + diff * this.scrollAnimationFraction,
+    );
+
+    requestAnimationFrame(this.stepScrollAnimation.bind(this));
+  }
+
+  /**
+   * Handles mouse wheel events.
+   *
+   * @param e The mouse wheel event to handle.
+   */
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  protected override wheel_(e: WheelEvent) {
+    // Don't scroll in response to mouse wheel events if we're currently
+    // animating scrolling to a category.
+    if (!this.scrollTarget) {
+      super.wheel_(e);
+    }
```

Comment:
Keep main flow outdented where possible by using guard clauses (especially appropriate here given wording of comment):

```suggestion
    // Don't scroll in response to mouse wheel events if we're currently
    // animating scrolling to a category.
    if (this.scrollTarget) return;
    super.wheel_(e);
```

Reply:
Done.